utils/ESR_Peers_List.cpp

```diff
@@ -0,0 +1,317 @@
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License, version 3,
+// as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+// The following curl will get a list of peers for a given swarm id:
+// curl --data '{"jsonrpc":"2.0","method":"eth_call","params":[{"to":"0x3e4BDd757322eF358664090c69b6927BdB6cf2De","data": "0x46e76d8b0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000d426c757a656c6c65537761726d00000000000000000000000000000000000000" },"latest"],"id":1}' https://ropsten.infura.io/uvek7IebbbHoP8Bb9NkV
+
+
+// Given a swarm id and a node address, the following curl will get the node info
+// curl --data '{"jsonrpc":"2.0","method":"eth_call","params":[{"to":"0x3e4BDd757322eF358664090c69b6927BdB6cf2De","data": "0xcc8575cb00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000d426c757a656c6c65537761726d00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c3230392e35302e36312e38370000000000000000000000000000000000000000" },"latest"],"id":1}' https://ropsten.infura.io/uvek7IebbbHoP8Bb9NkV
+
+#include <options/options.hpp>
+#include <utils/curl.hpp>
+#include <utils/ESR_Peers_List.h>
+#include <boost/algorithm/hex.hpp>
+#include <boost/algorithm/string.hpp>
+#include <json/json.h>
+#include <iostream>
+#include <sstream>
+#include <cassert>
+#include <sstream>
+
+namespace
+{
+    const std::string ERR_UNABLE_TO_PARSE_JSON_RESPONSE{"Unable to parse JSON response: "};
+
+    const std::string GET_NODE_LIST_ABI{R"({"constant": true, "inputs": [{"name": "swarmID", "type": "string"}], "name": "getNodeList", "outputs": [{"name": "", "type": "string[]"}], "payable": false, "stateMutability": "view", "type": "function", "signature": "0x46e76d8b"})"};
+    const std::string GET_PEER_INFO_ABI{R"({"constant": true, "inputs": [{"name": "swarmID", "type": "string"}, {"name": "nodeUUID", "type": "string"}], "name": "getNodeInfo", "outputs": [{"name": "nodeCount", "type": "uint256"}, {"name": "nodeHost","type": "string"}, {"name": "nodeHttpPort", "type": "uint256"}, {"name": "nodeName", "type": "string"}, {"name": "nodePort", "type": "uint256"}], "payable": false, "stateMutability": "view", "type": "function", "signature": "0xcc8575cb"})"};
+
+
+    void
+    trim_right_nulls(std::string& s)
```

Comment:
what's the point? It's only used once

Reply:
trim_right_nulls is used 3 times, may I keep it?